adapters/hong-kong.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * This code is responsible for implementing all methods related to fetching
+ * and returning data for the Hong Kong data source.
+ */
+'use strict';
+
+import { REQUEST_TIMEOUT } from '../lib/constants';
+import { default as baseRequest } from 'request';
+import cheerio from 'cheerio';
+import { default as moment } from 'moment-timezone';
+import { convertUnits } from '../lib/utils';
+const request = baseRequest.defaults({timeout: REQUEST_TIMEOUT});
+
+exports.name = 'hong-kong';
+
+/**
+ * Fetches the data for a given source and returns an appropriate object
+ * @param {object} source A valid source object
+ * @param {function} cb A callback of the form cb(err, data)
+ */
+exports.fetchData = (source, cb) => {
+  request(`${source.url}/24pc_Eng.xml`, function (err, res, body) {
+    if (err || res.statusCode !== 200) {}
+
+    try {
+      const data = formatData(body);
+
+      if (data.length === 0) {
+        return cb({message: 'Failure to parse data.'});
+      }
+      return cb(null, data);
+    } catch (e) {
+      return cb({message: 'Unknown adapter error.'});
+    }
+  });
+};
+
+/**
+ * Given fetched data, turn it into a format our system can use.
+ * @param {object} data Fetched source data and other metadata
+ * @return {object} Parsed and standarized data our system can use
+ */
+const formatData = function (data) {
+  let measurements = [];
+  let $ = cheerio.load(data, { xmlMode: true });
+  const rootElement = $('AQHI24HrPollutantConcentration');
+  // Gets the last build date for getting current measurements
+  const lastBuildDate = moment.tz(rootElement.children('lastBuildDate').text(), 'ddd, D MMM YYYY HH:mm:ss Z', 'Asia/Hong_Kong');
+
+  // Traverses the AQHI24HrPollutantConcentration > PollutantConcentration and
+  // filters the measurements in lastBuildDate
+  rootElement.children('PollutantConcentration').filter(function (index, element) {
```

Comment:
I like the `filter` here to cut down on number of measurements that try to get inserted. However, `moment` seems to have trouble parsing the date and so the filter wasn't working correctly. I spent a fair amount of time trying to troubleshoot, but couldn't figure it out. Could you change the logic so that the `DateTime` for each measurement is just checked against the current time (should probably use UTC for this check) and sees that it's less than 2 hours old?

Reply:
@jflasher Sorry for late reply. Since, we may not fetch duplicated measurements, I think the logic is filtered by `lastBuildDate` is better.

Reply:
I dont't find the parsing error in the filter when I'm performing a dry run. Could you paste the error message?

Reply:
No worries about the late reply, thanks for working on the adapter! Just to note, the adapters do not need to ensure uniqueness, the database will do that. So the adapter can grab duplicates and pass them along, but it's better if it doesn't try to insert alot of measurements that will likely be duplicates, and it adds database overheard to check for uniqueness and slows down inserts.

For the date piece, nevermind what I said! I wrote this nice long explanation and then said maybe it was just a version issue with `moment`. I upped the version and it works fine! Will change the package.json and merge in from a different branch. Thanks!